config/config.go

```diff
@@ -1304,3 +1308,15 @@ func getGoGCEnv() int {
 	}
 	return DefaultRuntimeConfig.GoGC
 }
+
+// OTLPConfig is the configuration for writing to the OTLP endpoint.
+type OTLPConfig struct {
+	PromoteResourceAttributes []string `yaml:"promote_resource_attributes,omitempty"`
+}
+
+// UnmarshalYAML implements the yaml.Unmarshaler interface.
+func (c *OTLPConfig) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	*c = DefaultOTLPConfig
+	type plain OTLPConfig
+	return unmarshal((*plain)(c))
```

Comment:
Should this check for duplicates? Or remove them?

Reply:
Good point @jesusvazquez. Maybe just remove them?

Reply:
I modified the code to sanitize (trim whitespace, drop duplicates) the configured resource attribute set. PTAL.